aspnetcore/blazor/state-management.md

```diff
@@ -0,0 +1,357 @@
+
+To recover the `currentCount` data if the user returns to the `Counter` component later (including if they're on an entirely new circuit), use `ProtectedSessionStore.GetAsync`:
+
+```csharp
+protected override async Task OnInitializedAsync()
+{
+    currentCount = await ProtectedSessionStore.GetAsync<int>("count");
+}
+```
+
+If the component's parameters include navigation state, call `ProtectedSessionStore.GetAsync` and assign the result in `OnParametersSetAsync`, not `OnInitializedAsync`. `OnInitializedAsync` is only called once when the component is first instantiated and isn't called again later if the user navigates to a different URL while remaining on the same page.
+
+> [!WARNING]
+> The examples in this section only work if the server doesn't have prerendering enabled. With prerendering enabled, an error is generated similar to:
+>
+> > JavaScript interop calls cannot be issued at this time. This is because the component is being prerendered.
+>
+> Either disable prerendering or add additional code to work with prerendering, as described in the [Handle prerendering](#handle-prerendering) section.
+
+### Handle the loading state
+
+Since browser storage is asynchronous (accessed over a network connection), there's always a period of time before the data is loaded and available for use by a component. For the best results, render a loading-state message while loading is in progress instead of displaying blank or default data.
+
+One approach is to track whether the data is `null` (still loading) or not. In the default `Counter` component, the count is held in an `int`. Make `currentCount` nullable by adding a question mark (`?`) to the type (`int`):
+
+```csharp
+private int? currentCount;
+```
+
+Instead of displaying the count and **Increment** button unconditionally, choose to display these elements only if the data is loaded:
+
+```cshtml
+@if (currentCount.HasValue)
+{
+    <p>Current count: <strong>@currentCount</strong></p>
+
+    <button @onclick="IncrementCount">Increment</button>
+}
+else
+{
+    <p>Loading...</p>
+}
+```
+
+### Handle prerendering
+
+During prerendering, an interactive connection to the user's browser doesn't exist, and the browser doesn't yet have a page in which it can run JavaScript code. It's not possible to interact with `localStorage` or `sessionStorage` at this time. If the component attempts to interact with storage, an error is generated similar to:
+
+> JavaScript interop calls cannot be issued at this time. This is because the component is being prerendered.
+
+One way to resolve the error is to disable prerendering. This is usually the best choice if the app makes heavy use of browser-based storage. Prerendering adds complexity and doesn't benefit the app because the app can't prerender any useful content until `localStorage` or `sessionStorage` are available.
+
+To disable prerendering, open the *Pages/_Host.cshtml* file and remove the call to `Html.RenderComponentAsync`.
```

Comment:
This text will likely need to be modified for Preview 8 / 9.

Reply:
I'll open a tracking issue for it.